### PR TITLE
:memo: feat(docs): New diagram for the development workflow

### PR DESCRIPTION
**Pull request by Michiels96**
# Pull request for Candy-doc

## Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../../CONTRIBUTING.md) to
this repository.

- [x] Make sure you are requesting to **pull a feature/bugfix branch** (right
  side). Don't request your master!
- [x] Make sure you are making a pull request against the **main branch** (left
  side).
- [x] Check the commit's or even all commits' message styles matches our
  requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit
  test.

## Description

The actual diagram of the development workflow is deprecated.
In order to improve the understanding of the development workflow and to have a versionning system of this one.

More info on issue https://github.com/Candy-Doc/candy-doc-maven-plugin/issues/135

❤️ Thank you!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -210,7 +210,29 @@ _main_ and merged to this latter once the feature is ready.
 Every time a new version is about to be released, a merge from your new branch to _main_
 will create a new tag and a new package in the package registry.
```
Comment (Tifloz) [inline README.md:212]: Can you add a quick explanation of what your diagrams show us please ?

**Review by Tifloz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Michiels96) [reply README.md:209]: Done

**Comment by Michiels96** [comment]
Hey https://github.com/all-contributors can you add @Michiels96 for review please ?

**Review by Tifloz — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -221,13 +220,33 @@ milestones [here](https://github.com/Candy-Doc/candy-doc-maven-plugin/milestones
...
+    1 --> 2
+    2 --> 3
+    3 --> 4
+    1 --feat/my-feature--> 2bis
+    2bis --> 3bis
+    3bis --> 4bis
+```
 
-Every time a new version is about to be released, a merge from your new branch to _main_
+Then, every time a new version is about to be released, a rebase from your new branch (feat/<new_branch>) to _main_
```
Comment (ctacquet) [inline README.md:237]: This is not true, rebase doesn't trigger a release or tag creation

File: README.md [hunk 3]
```diff
@@ -221,13 +220,33 @@ milestones [here](https://github.com/Candy-Doc/candy-doc-maven-plugin/milestones
 
 The development team works on a branch named _main_.
 
-Whenever a feature needs to be implemented, a new branch must be pulled from
-_main_ and merged to this latter once the feature is ready.
+Whenever a feature needs to be implemented, a new branch (feat/<new_branch>) must be pulled from
+_main_ and rebased to this latter once the feature is ready.
```
Comment (ctacquet) [inline README.md:224]: Later not latter

**Review by ctacquet — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (Michiels96) [reply README.md]: Done.

File: README.md (hunk 3 above)
Comment (Michiels96) [reply README.md]: Checked.

**Review by ctacquet — APPROVED** [review]

**Review by Tifloz — APPROVED** [review]

**Pull request merged**